app/src/main/java/com/android/bookswap/model/map/Geolocation.kt

```diff
@@ -43,24 +48,24 @@ class Geolocation(private val activity: Activity) {
   private val locationCallback =
       object : LocationCallback() {
         override fun onLocationResult(p0: LocationResult) {
-          p0.lastLocation?.let { location ->
+          p0.lastLocation.let { location ->
```

Comment:
We concluded in a previous PR that it would be better to check lastLocation, even if the IDE says it's unnecessary.

Reply:
LocationCallback specify the `@NonNull` annotation denoting that the parameter can never be null. We can thus safely remove the '?'

```kotlin
public abstract class LocationCallback {
    public LocationCallback() {
    }

    public void onLocationResult(@NonNull LocationResult var1) {
    }

    public void onLocationAvailability(@NonNull LocationAvailability var1) {
    }
}
```